shared-module/displayio/OnDiskBitmap.c

```diff
@@ -141,38 +200,87 @@ uint32_t common_hal_displayio_ondiskbitmap_get_pixel(displayio_ondiskbitmap_t *s
         uint8_t red;
         uint8_t green;
         uint8_t blue;
-        if (self->bits_per_pixel == 1) {
+
+        if (self->dither) {
+            uint8_t noise =  noise2d(x,y) * 255;
+            uint8_t randr  = noise & self->dither_mask_r;
+            uint8_t randg  = (noise >> 3) & self->dither_mask_g;
+            uint8_t randb  = (noise >> 6) & self->dither_mask_b;
+
+            if (self->bits_per_pixel == 1) {
+                uint8_t bit_offset = x%8;
+                tmp = ( pixel_data & (0x80 >> (bit_offset))) >> (7 - bit_offset);
+                if (tmp == 1) {
+                    return 0x00FFFFFF;
+                } else {
+                    return 0x00000000;
+                }
+            } else if (bytes_per_pixel == 1) {
+                blue = MIN(255,((self->palette_data[pixel_data] & 0xFF) >> 0) + randb);
+                red = MIN(255,((self->palette_data[pixel_data] & 0xFF0000) >> 16) + randr);
+                green = MIN(255,((self->palette_data[pixel_data] & 0xFF00) >> 8) + randg);
+                tmp = (red << 16 | green << 8 | blue );
+                return tmp;
+            } else if (bytes_per_pixel == 2) {
+                if (self->g_bitmask == 0x07e0) { // 565
+                    red =((pixel_data & self->r_bitmask) >>11);
+                    green = ((pixel_data & self->g_bitmask) >>5);
+                    blue = ((pixel_data & self->b_bitmask) >> 0);
+                } else { // 555
+                    red =((pixel_data & self->r_bitmask) >>10);
+                    green = ((pixel_data & self->g_bitmask) >>4);
+                    blue = ((pixel_data & self->b_bitmask) >> 0);
+                }
+                tmp = (red << 19 | green << 10 | blue << 3);
+                return tmp;
+            } else if ((bytes_per_pixel == 4) && (self->bitfield_compressed)) {
+                blue = MIN(255, (pixel_data & 0xFF) + randb);
+                green = MIN(255, ((pixel_data >> 8) & 0xFF) + randg);
+                red = MIN(255,((pixel_data >> 16) & 0xFF)  + randr);
+
+                return (red << 16 | green << 8 | blue );
+            } else {
+                blue = MIN(255, (pixel_data & 0xFF) + randb);
+                green = MIN(255, ((pixel_data >> 8) & 0xFF) + randg);
+                red = MIN(255,((pixel_data >> 16) & 0xFF)  + randr);
+
+                return (red << 16 | green << 8 | blue );
+            }
+        } else { // No dither
```

Comment:
What if you assigned randb/randg/randr all 0 when "no dither" is selected?  This would improve code size, but possibly slow down the !dither case a little bit.  Does the slow-down matter?